vllm/config.py

```diff
@@ -969,16 +970,27 @@ def _maybe_override_draft_max_model_len(
 
     @staticmethod
     def create_draft_parallel_config(
-            target_parallel_config: ParallelConfig) -> ParallelConfig:
+            target_parallel_config: ParallelConfig,
+            speculative_tensor_parallel_size: Optional[int]) -> ParallelConfig:
         """Create a parallel config for use by the draft worker.
 
-        This is mostly a copy of the target parallel config. In the future the
-        draft worker can have a different parallel strategy, e.g. TP=1.
+        This is mostly a copy of the target parallel config, except the tp_size.
         """
+
+        speculative_tensor_parallel_size = (
+            speculative_tensor_parallel_size
+            or target_parallel_config.tensor_parallel_size)
+
+        if speculative_tensor_parallel_size > \
+            target_parallel_config.tensor_parallel_size:
+            raise ValueError(
+                f"{speculative_tensor_parallel_size=} cannot be "
+                f"larger than {target_parallel_config.tensor_parallel_size}")
```

Comment:
nit: should keep the validation in `SpeculativeConfig._verify_args`

Reply:
I tried that, but it was impossible because I had to compare it to target_parallel_config, which is not SpeculativeConfig itself.

So, I followed the style in `_maybe_override_draft_max_model_len()`.

Reply:
If you have any ideas for cleanup, please let me know :)

Reply:
I see -- this is good, thanks